repo/fsrepo/fsrepo.go

```diff
@@ -495,19 +495,52 @@ func (r *FSRepo) SetConfigKey(key string, value interface{}) error {
 	if err != nil {
 		return err
 	}
-	switch v := value.(type) {
-	case string:
-		if i, err := strconv.Atoi(v); err == nil {
-			value = i
-		}
-	}
 	var mapconf map[string]interface{}
 	if err := serialize.ReadConfigFile(filename, &mapconf); err != nil {
 		return err
 	}
-	if err := common.MapSetKV(mapconf, key, value); err != nil {
+
+	// Get the type of the value associated with the key
+	oldValue, err := common.MapGetKV(mapconf, key)
+	var castedValue interface{}
+	ok := true
+	if err != nil {
+		// key-value does not exist yet
+		v, _ := value.(string)
+		castedValue, err = strconv.ParseBool(v)
+		if err != nil {
+			castedValue, err = strconv.Atoi(v)
+			if err != nil {
+				castedValue, err = strconv.ParseFloat(v, 32)
+				if err != nil {
+					castedValue = v
+				}
+			}
+		}
+	} else {
+		switch oldValue.(type) {
+		case bool:
+			castedValue, ok = value.(bool)
+		case int:
+			castedValue, ok = value.(int)
+		case float32:
+			castedValue, ok = value.(float32)
+		case string:
+			castedValue, ok = value.(string)
+		default:
+			castedValue = value
+		}
+		if !ok {
+			return fmt.Errorf("Wrong config type, expected %T", oldValue)
```

Comment:
This entire switch-case can be moved to common if we expect the values in the "json object" to not change in type. Wish could be cleaner.